core/trino-main/src/main/java/io/trino/server/ServerMainModule.java

```diff
@@ -207,6 +208,8 @@ protected void setup(Binder binder)
             httpServerConfig.setAdminEnabled(false);
         });
 
+        // TODO move PreparedStatementEncoder to CoordinatorModule
```

Comment:
why can this be not be done now? Because of `ProtocolConfig` being binded in ServerMainModule too?

Reply:
PreparedStatementEncoder  is a dependency of HttpRequestSessionContextFactory 
HttpRequestSessionContextFactory is a dependency of ResourceSecurityDynamicFeature
ResourceSecurityDynamicFeature is needed on also on workers

will remove the TODO